### ISSUE TITLE
Fix All in Document forces binding of whole document

### ISSUE THREAD

**Issue by Therzok** [mono/monodevelop#4382]
When querying the fix all menu, we end up binding the whole document. Fix All in Document should be deffered.

```
0.49%   <CreateFixMenu>b__0  •  63,102* ms  •  8 calls  •  MonoDevelop.CodeActions.CodeFixMenuService+<>c__DisplayClass1_2.<CreateFixMenu>b__0
  0.28%   GetFixAsync  •  36,485* ms  •  5 calls  •  Microsoft.CodeAnalysis.CodeFixes.SyntaxEditorBasedCodeFixProvider+SyntaxEditorBasedFixAllProvider.GetFixAsync(FixAllContext)
    0.28%   Start  •  36,485* ms  •  5 calls  •  System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.Start(TStateMachine&)
      0.28%   MoveNext  •  36,485* ms  •  5 calls  •  Microsoft.CodeAnalysis.CodeFixes.SyntaxEditorBasedCodeFixProvider+SyntaxEditorBasedFixAllProvider+<GetFixAsync>d__2.MoveNext
        0.28%   GetDocumentDiagnosticsToFixAsync  •  36,485* ms  •  5 calls  •  Microsoft.CodeAnalysis.CodeFixes.FixAllContext.GetDocumentDiagnosticsToFixAsync
          0.28%   GetDocumentDiagnosticsToFixAsync  •  36,485* ms  •  5 calls  •  Microsoft.CodeAnalysis.CodeFixes.FixAllContext+DiagnosticProvider.GetDocumentDiagnosticsToFixAsync(FixAllContext)
            0.28%   Start  •  36,485* ms  •  5 calls  •  System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.Start(TStateMachine&)
              0.28%   MoveNext  •  36,485* ms  •  5 calls  •  Microsoft.CodeAnalysis.CodeFixes.FixAllContext+DiagnosticProvider+<GetDocumentDiagnosticsToFixAsync>d__5.MoveNext
                ►0.28%   GetDocumentDiagnosticsToFixWorkerAsync  •  36,485* ms  •  5 calls  •  Microsoft.CodeAnalysis.CodeFixes.FixAllContext+DiagnosticProvider.GetDocumentDiagnosticsToFixWorkerAsync(FixAllContext)
                ►0.00%   AwaitUnsafeOnCompleted  •  0 ms  •  1 call  •  System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AwaitUnsafeOnCompleted(TAwaiter&, TStateMachine&)
              ►0.00%   Undo  •  0 ms  •  1 call  •  System.Threading.ExecutionContextSwitcher.Undo
               0.00%   EstablishCopyOnWriteScope  •  0 ms  •  2 calls  •  System.Threading.ExecutionContext.EstablishCopyOnWriteScope(Thread, Boolean, ExecutionContextSwitcher&)
             0.00%   get_Task  •  0 ms  •  1 call  •  System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.get_Task
        ►0.00%   AwaitUnsafeOnCompleted  •  0 ms  •  1 call  •  System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AwaitUnsafeOnCompleted(TAwaiter&, TStateMachine&)
      ►0.00%   Undo  •  0 ms  •  1 call  •  System.Threading.ExecutionContextSwitcher.Undo
       0.00%   EstablishCopyOnWriteScope  •  0 ms  •  2 calls  •  System.Threading.ExecutionContext.EstablishCopyOnWriteScope(Thread, Boolean, ExecutionContextSwitcher&)
     0.00%   get_Task  •  0 ms  •  1 call  •  System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.get_Task
  0.21%   GetFixAsync  •  26,617 ms  •  3 calls  •  Microsoft.CodeAnalysis.CodeFixes.BatchFixAllProvider.GetFixAsync(FixAllContext)
     0.00%   128 functions hidden  •  20 ms total  •  7943 calls total
   0.00%   2 functions hidden  •  0 ms total  •  6 calls total
```

> VS bug [#591958](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/591958)

**Comment by Therzok** [comment]
https://github.com/mono/monodevelop/pull/4431

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-18] Comment by **xamarinc**



Fixed in version 8.6.0.802 (master)



Author: therzok  
 Commit: c5ec9d8c7551ce6a1c7ec6b574621b5416568d13 (xamarin/monodevelop)
